app/src/main/java/org/simple/clinic/bloodsugar/entry/BloodSugarEntryUiRenderer.kt

```diff
@@ -15,6 +19,15 @@ class BloodSugarEntryUiRenderer(
     openAsValueChangedCallback.pass(model.openAs) { setupUi(it) }
     manageProgress(model)
     manageBloodSugarUnitPreferenceButtonText(model)
+    manageBloodSugarUnitPreferenceButtonVisibility(model)
+  }
+
+  private fun manageBloodSugarUnitPreferenceButtonVisibility(model: BloodSugarEntryModel) {
+    when (model.openAs.measurementType) {
+      Random, PostPrandial, Fasting -> ui.showBloodSugarUnitPreferenceButton()
+      HbA1c -> ui.hideBloodSugarUnitPreferenceButton()
+      else -> ui.showBloodSugarUnitPreferenceButton()
```

Comment:
It doesn't seem like we are hiding the blood sugar unit button for unknow measurement type, please check? Also let's use `Unknown` instead of an else block.

Reply:
Hey, Can you check this again? I can see this has been updated. I'll update it to use `Unknown` instead of `else`

Reply:
Oh okay, I missed it. you have changed it a different commit where we are handling decimal input.